src/main/java/teammates/common/datatransfer/questions/FeedbackMcqQuestionDetails.java

```diff
@@ -95,19 +116,75 @@ public boolean extractQuestionDetails(
                 }
             }
 
-            setMcqQuestionDetails(numOfMcqChoices, mcqChoices, mcqOtherEnabled);
+            String hasAssignedWeightsString = HttpRequestHelper.getValueFromParamMap(
+                    requestParameters, Const.ParamsNames.FEEDBACK_QUESTION_MCQ_WEIGHTS_ASSIGNED);
+            boolean hasAssignedWeights = "on".equals(hasAssignedWeightsString);
+            List<Double> mcqWeights = getMcqWeights(
+                    requestParameters, numMcqChoicesCreated, hasAssignedWeights);
+            double mcqOtherWeight = getMcqOtherWeight(requestParameters, mcqOtherEnabled, hasAssignedWeights);
+            setMcqQuestionDetails(
+                    numOfMcqChoices, mcqChoices, mcqOtherEnabled, hasAssignedWeights, mcqWeights, mcqOtherWeight);
         } else {
             setMcqQuestionDetails(FeedbackParticipantType.valueOf(generatedMcqOptions));
         }
         return true;
     }
 
-    private void setMcqQuestionDetails(int numOfMcqChoices,
-            List<String> mcqChoices,
-            boolean otherEnabled) {
+    private List<Double> getMcqWeights(Map<String, String[]> requestParameters,
+            int numMcqChoicesCreated, boolean hasAssignedWeights) {
+        List<Double> mcqWeights = new ArrayList<>();
+
+        if (!hasAssignedWeights) {
+            return mcqWeights;
+        }
+
+        for (int i = 0; i < numMcqChoicesCreated; i++) {
+            String choice = HttpRequestHelper.getValueFromParamMap(
+                    requestParameters, Const.ParamsNames.FEEDBACK_QUESTION_MCQCHOICE + "-" + i);
+            String weight = HttpRequestHelper.getValueFromParamMap(
+                    requestParameters, Const.ParamsNames.FEEDBACK_QUESTION_MCQ_WEIGHT + "-" + i);
+
+            if (choice != null && !choice.trim().isEmpty() && weight != null) {
```

Comment:
In case if the user enters all spaces `"     "`.
The `extractQuestionDetails` method eliminates such choices from the list.
```java
                if (mcqChoice != null && !mcqChoice.trim().isEmpty()) {
                    mcqChoices.add(mcqChoice);
                    numOfMcqChoices++;
                }

```
 so I am also doing the same thing here. As, there is no point parsing the weight of a empty choice as it will anyway not pass validation.

Reply:
That's the validation logic rather than extraction logic I would say if we want clear definition between extractXXX and ValidateXXX.

Reply:
>That's the validation logic rather than extraction logic I would say if we want clear definition between extractXXX and ValidateXXX.

@xpdavid Sorry, I didn't understand this. :sweat_smile:

Reply:
Take a look at `extractQuestionDetails` and `validateQuestionDetails`.
I think It's OK to leave it there.